File: ciam/src/main/java/com/crm/ciam/seed/UserSeeder.java
package com.crm.ciam.seed;

import java.util.Set;

import org.springframework.boot.CommandLineRunner;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import com.crm.ciam.entity.Benutzer;
import com.crm.ciam.entity.enums.BenutzerRolle;
import com.crm.ciam.repository.BenutzerRepository;

@Component
public class UserSeeder implements CommandLineRunner {

    private final BenutzerRepository benutzerRepository;
    private final PasswordEncoder passwordEncoder;

    public UserSeeder(BenutzerRepository benutzerRepository, PasswordEncoder passwordEncoder) {
        this.benutzerRepository = benutzerRepository;
        this.passwordEncoder = passwordEncoder;
    }

    @Override
    @Transactional
    public void run(String... args) {
        if (benutzerRepository.count() > 0) return;

        Benutzer admin = new Benutzer();
        admin.setBenutzername("admin");
        admin.setPasswort(passwordEncoder.encode("admin123"));
        admin.setVorname("Admin");
        admin.setNachname("System");
        admin.setEmail("admin@crm.local");
        admin.setRollen(Set.of(BenutzerRolle.ADMIN));
        benutzerRepository.save(admin);

        Benutzer vertrieb = new Benutzer();
        vertrieb.setBenutzername("vertrieb");
        vertrieb.setPasswort(passwordEncoder.encode("test123"));
        vertrieb.setVorname("Vera");
        vertrieb.setNachname("Vertrieb");
        vertrieb.setEmail("vertrieb@crm.local");
        vertrieb.setRollen(Set.of(BenutzerRolle.VERTRIEB));
        benutzerRepository.save(vertrieb);

        Benutzer personal = new Benutzer();
        personal.setBenutzername("personal");
        personal.setPasswort(passwordEncoder.encode("test123"));
        personal.setVorname("Paul");
        personal.setNachname("Personal");
        personal.setEmail("personal@crm.local");
        personal.setRollen(Set.of(BenutzerRolle.PERSONAL));
        benutzerRepository.save(personal);

        Benutzer allrounder = new Benutzer();
        allrounder.setBenutzername("allrounder");
        allrounder.setPasswort(passwordEncoder.encode("test123"));
        allrounder.setVorname("Alex");
        allrounder.setNachname("Allrounder");
        allrounder.setEmail("allrounder@crm.local");
        allrounder.setRollen(Set.of(BenutzerRolle.VERTRIEB, BenutzerRolle.PERSONAL));
        benutzerRepository.save(allrounder);

        Benutzer demo = new Benutzer();
        demo.setBenutzername("demo");
        demo.setPasswort(passwordEncoder.encode("demo1234"));
        demo.setVorname("David");
        demo.setNachname("Demo");
        demo.setEmail("demo@crm.local");
        demo.setRollen(Set.of(BenutzerRolle.ADMIN));
        benutzerRepository.save(demo);

        System.out.println("=== CIAM UserSeeder: 5 Benutzer erstellt ===");
    }
}

File: ciam/src/main/kotlin/com/crm/ciam/seed/UserSeeder.kt
package com.crm.ciam.seed

import com.crm.ciam.entity.Benutzer
import com.crm.ciam.entity.enums.BenutzerRolle
import com.crm.ciam.repository.BenutzerRepository
import org.springframework.boot.CommandLineRunner
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
class UserSeeder(
    private val benutzerRepository: BenutzerRepository,
    private val passwordEncoder: PasswordEncoder
) : CommandLineRunner {

    @Transactional
    override fun run(vararg args: String) {
        if (benutzerRepository.count() > 0) return

        val admin = Benutzer().apply {
            benutzername = "admin"
            passwort = passwordEncoder.encode("admin123")
            vorname = "Admin"
            nachname = "System"
            email = "admin@crm.local"
            rollen = mutableSetOf(BenutzerRolle.ADMIN)
        }
        benutzerRepository.save(admin)

        val vertrieb = Benutzer().apply {
            benutzername = "vertrieb"
            passwort = passwordEncoder.encode("test123")
            vorname = "Vera"
            nachname = "Vertrieb"
            email = "vertrieb@crm.local"
            rollen = mutableSetOf(BenutzerRolle.VERTRIEB)
        }
        benutzerRepository.save(vertrieb)

        val personal = Benutzer().apply {
            benutzername = "personal"
            passwort = passwordEncoder.encode("test123")
            vorname = "Paul"
            nachname = "Personal"
            email = "personal@crm.local"
            rollen = mutableSetOf(BenutzerRolle.PERSONAL)
        }
        benutzerRepository.save(personal)

        val allrounder = Benutzer().apply {
            benutzername = "allrounder"
            passwort = passwordEncoder.encode("test123")
            vorname = "Alex"
            nachname = "Allrounder"
            email = "allrounder@crm.local"
            rollen = mutableSetOf(BenutzerRolle.VERTRIEB, BenutzerRolle.PERSONAL)
        }
        benutzerRepository.save(allrounder)

        val demo = Benutzer().apply {
            benutzername = "demo"
            passwort = passwordEncoder.encode("demo1234")
            vorname = "David"
            nachname = "Demo"
            email = "demo@crm.local"
            rollen = mutableSetOf(BenutzerRolle.ADMIN)
        }
        benutzerRepository.save(demo)

        println("=== CIAM UserSeeder: 5 Benutzer erstellt ===")
    }
}
